samples/greengrass/basic_discovery/main.cpp

```diff
@@ -190,10 +190,8 @@ int main(int argc, char *argv[])
     Aws::Iot::MqttClient mqttClient(bootstrap);
     std::shared_ptr<Mqtt::MqttConnection> connection(nullptr);
 
-    std::mutex semaphoreLock;
-    std::condition_variable semaphore;
-    std::atomic<bool> connectionFinished(false);
-    std::atomic<bool> shutdownCompleted(false);
+    std::promise<bool> connectionFinishedPromise;
+    std::promise<bool> shutdownCompletedPromise;
```

Comment:
trivial: these can be `std::promise<void>` since we don't do anything with the bool inside them